File: leetcode/2215.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {

        set<int> nums1_set, nums2_set;
        vector<vector<int>> answer(2,vector<int>());

        for(auto i : nums1){
            nums1_set.insert(i);
        }

        for(auto i : nums2){
            nums2_set.insert(i);
        }

        for(auto num: nums1_set){
            if (nums2_set.find(num) == nums2_set.end()){
                answer[0].push_back(num);
            }
        }

        for(auto num: nums2_set){
            if (nums1_set.find(num) == nums1_set.end()){
                answer[1].push_back(num);
            }
        }

        return answer;
    }
};

File: leetcode/2215.go
func findDifference(nums1 []int, nums2 []int) [][]int {

	m1 := make(map[int]int);
	m2 := make(map[int]int);
	var answer [][]int
	var answer0 []int
	var answer1 []int

	for _, val := range nums1 {
		m1[val] = 1
	}
	for _, val := range nums2 {
		m2[val] = 1
	}

	for key, _ := range m1 {
		if _, ok := m2[key]; !ok {
			answer0 = append(answer0, key)
		}
	}
	for key, _ := range m2 {
		if _, ok := m1[key]; !ok {
			answer1 = append(answer1, key)
		}
	}
	answer = append(answer, answer0)
	answer = append(answer, answer1)
	return answer
}